components/tikv_util/src/yatp_pool/mod.rs

```diff
@@ -0,0 +1,230 @@
+// Copyright 2020 TiKV Project Authors. Licensed under Apache-2.0.
+
+mod future_pool;
+mod metrics;
+pub use future_pool::{Full, FuturePool};
+
+use std::sync::Arc;
+use yatp::pool::{CloneRunnerBuilder, Local, Runner};
+use yatp::queue::{multilevel, QueueType};
+use yatp::task::future::{Runner as FutureRunner, TaskCell};
+use yatp::ThreadPool;
+
+pub trait PoolTicker: Send + Clone + 'static {
+    fn on_tick(&mut self);
```

Comment:
The name `on_tick` is not so correct. This method is not executed every n seconds but is called after every `handle`.

I think a real ticker should contain a tick interval and `on_tick` automatically calls after each interval...

Reply:
So shall we add one method `fn get_tick_interval()` ? Or rename the method `on_tick` to `maybe_tick`?

Reply:
I hope we don't implement the tick logic in every ticker, so I prefer the former.